app/models/work_package/validations.rb

```diff
@@ -34,7 +34,7 @@ module WorkPackage::Validations
 
     validates :subject, length: { maximum: 255 }
     validates :done_ratio, inclusion: { in: 0..100 }, allow_nil: true
-    validates :estimated_hours, numericality: { allow_nil: true }
+    validates :estimated_hours, numericality: { allow_nil: true, greater_than_or_equal_to: 0 }
```

Comment:
Great add. I checked the table's constraints and it seems these are not being enforced at the database layer. Do we want to keep this enforced at the application layer only?

```sql
           constraint_name           | constraint_type |                                      definition
-------------------------------------+-----------------+--------------------------------------------------------------------------------------
 work_packages_due_larger_start_date | c               | CHECK ((due_date >= start_date))
 work_packages_pkey                  | p               | PRIMARY KEY (id)
 fk_rails_931ad309e8                 | f               | FOREIGN KEY (project_id) REFERENCES projects(id) ON UPDATE CASCADE ON DELETE CASCADE
 fk_rails_f2a8977aa1                 | f               | FOREIGN KEY (type_id) REFERENCES types(id) ON UPDATE CASCADE ON DELETE CASCADE
 fk_rails_5edb6f06e6                 | f               | FOREIGN KEY (status_id) REFERENCES statuses(id) ON UPDATE CASCADE ON DELETE CASCADE
(5 rows)
```

Reply:
Good point. Yes probably, but I don't know.
I was only moving this positive check on `estimated_hours` from the contract to the model so it's done in the same way as `remaining_hours` and the error messages are the same. They were different before.
I'd like to discuss that in a dev discussion. Contraints could be added for all 3 progress fields.